tests/unit/schema/test_api.py

```diff
@@ -1287,3 +1289,224 @@ def test_map_field_name_disambiguation():
+                            "Giant": gapic_metadata_pb2.GapicMetadata.MethodList(methods=["giant"]),
+                            "Humboldt": gapic_metadata_pb2.GapicMetadata.MethodList(methods=["humboldt"]),
+                            "Ramshorn": gapic_metadata_pb2.GapicMetadata.MethodList(methods=["ramshorn"]),
+                        },
+                    ),
+
+                }
+            ),
+        }
+    )
+    actual = api_schema.gapic_metadata(opts)
+    assert expected == actual
+    expected = MessageToJson(expected, sort_keys=True)
+    actual = api_schema.gapic_metadata_json(opts)
+    assert expected == actual
+
+    opts = Options.build("transport=rest+grpc")
+    expected = gapic_metadata_pb2.GapicMetadata(
+        schema="1.0",
+        comment="This file maps proto services/RPCs to the corresponding library clients/methods",
+        language="python",
+        proto_package="animalia.mollusca.v1",
+        library_package="animalia.mollusca_v1",
+        services={
+            "Octopus": gapic_metadata_pb2.GapicMetadata.ServiceForTransport(
+                clients={
+                    "grpc": gapic_metadata_pb2.GapicMetadata.ServiceAsClient(
+                        library_client="OctopusClient",
+                        rpcs={
+                            "BlueSpot": gapic_metadata_pb2.GapicMetadata.MethodList(methods=["blue_spot"]),
+                            "GiantPacific": gapic_metadata_pb2.GapicMetadata.MethodList(methods=["giant_pacific"]),
+                        },
+                    ),
+                    "grpcAsync": gapic_metadata_pb2.GapicMetadata.ServiceAsClient(
+                        library_client="OctopusAsyncClient",
+                        rpcs={
+                            "BlueSpot": gapic_metadata_pb2.GapicMetadata.MethodList(methods=["blue_spot"]),
+                            "GiantPacific": gapic_metadata_pb2.GapicMetadata.MethodList(methods=["giant_pacific"]),
+                        },
+                    ),
+                    "rest": gapic_metadata_pb2.GapicMetadata.ServiceAsClient(
+                        library_client="OctopusClient",
+                        rpcs={
+                            "BlueSpot": gapic_metadata_pb2.GapicMetadata.MethodList(methods=["blue_spot"]),
+                            "GiantPacific": gapic_metadata_pb2.GapicMetadata.MethodList(methods=["giant_pacific"]),
+                        },
+                    )
+                }
+            ),
+            "Squid": gapic_metadata_pb2.GapicMetadata.ServiceForTransport(
+                clients={
+                    "grpc": gapic_metadata_pb2.GapicMetadata.ServiceAsClient(
+                        library_client="SquidClient",
+                        rpcs={
+                            "Giant": gapic_metadata_pb2.GapicMetadata.MethodList(methods=["giant"]),
+                            "Humboldt": gapic_metadata_pb2.GapicMetadata.MethodList(methods=["humboldt"]),
```

Comment:
Can you ever have more than one Python method for a given RPC? If so, it might be good to test that, as well.

Reply:
Per library client, no. As addressed elsewhere, asynchronous variants belong to a separate client class.